scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,472 @@
+            self.to_device(device)
+
+            if i == 0:
+                obs_plate_sites = self._get_obs_plate_sites(args, kwargs)
+                if len(obs_plate_sites) == 0:
+                    # if no local variables - don't sample
+                    break
+                obs_plate_dim = list(obs_plate_sites.values())[0]
+
+                sample_kwargs_obs_plate = sample_kwargs.copy()
+                sample_kwargs_obs_plate[
+                    "return_sites"
+                ] = self._get_obs_plate_return_sites(
+                    sample_kwargs, list(obs_plate_sites.keys())
+                )
+                sample_kwargs_obs_plate["show_progress"] = False
+
+                samples = self._get_posterior_samples(
+                    args, kwargs, **sample_kwargs_obs_plate
+                )
+            else:
+                samples_ = self._get_posterior_samples(
+                    args, kwargs, **sample_kwargs_obs_plate
+                )
+
+                samples = {
+                    k: np.array(
+                        [
+                            np.concatenate(
+                                [samples[k][j], samples_[k][j]],
+                                axis=obs_plate_dim,
+                            )
+                            for j in range(
+                                len(samples[k])
+                            )  # for each sample (in 0 dimension
+                        ]
+                    )
+                    for k in samples.keys()  # for each variable
+                }
+            i += 1
+
+        # sample global parameters
+        global_samples = self._get_posterior_samples(args, kwargs, **sample_kwargs)
+        global_samples = {
+            k: v
+            for k, v in global_samples.items()
+            if k not in list(obs_plate_sites.keys())
+        }
+
+        for k in global_samples.keys():
+            samples[k] = global_samples[k]
+
+        self.module.to(device)
+
+        return samples
+
+    def sample_posterior(
+        self,
+        num_samples: int = 1000,
+        return_sites: Optional[list] = None,
+        use_gpu: bool = None,
+        batch_size: int = 128,
+        sample_kwargs=None,
+        return_samples: bool = False,
+        summary_fun: Optional[dict] = None,
+    ):
+        """
+        Summarise posterior distribution.
+
+        Generate samples from posterior distribution for each parameter
+        and compute mean, 5%/95% quantiles, standard deviation.
+
+        Parameters
+        ----------
+        num_samples
+            number of posterior samples to generate.
+        return_site
+            get samples for pyro model variable, default is all variables, otherwise list variable names).
+        use_gpu
+            Load model on default GPU if available (if None or True),
+            or index of GPU to use (if int), or name of GPU (if str), or use CPU (if False).
+        sample_kwargs
```

Comment:
instead of sample_kwargs, can we just enumerate the parameters here? Users won't go looking at a private method and it doesn't seem like many params?